cmd/kubeadm/app/cmd/init.go

```diff
@@ -336,8 +339,7 @@ func (i *Init) Run(out io.Writer) error {
 		"KubeConfigName": kubeadmconstants.AdminKubeConfigFileName,
 		"Token":          i.cfg.Token,
 		"CAPubKeyPin":    pubkeypin.Hash(caCert),
-		"MasterIP":       i.cfg.API.AdvertiseAddress,
-		"MasterPort":     strconv.Itoa(int(i.cfg.API.BindPort)),
+		"MasterEndpoint": strings.TrimPrefix(masterEndpoint, "https://"),
```

Comment:
I think this approach is easier to read and understand as opposed to wrapping functions. My approach is similar to the original approach of creating the MasterPort, which converts the type. Additionally, all calls to the function, except this one do the same thing so wrapping functions feels like overkill. In a follow-on PR I plan to add a validation function that verifies that DiscoveryTokenAPIServers contains a valid IP:Port pair for v4 and v6. That should address the issue that we experienced with the DiscoveryTokenAPIServers containing "https". Please let me know if this is unacceptable to you and I will look into rewriting the GetMasterEndpoint function.